File: backend/src/main/java/de/grimsi/gameyfin/repositories/LibraryRepository.java
package de.grimsi.gameyfin.repositories;

import de.grimsi.gameyfin.entities.Library;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface LibraryRepository extends JpaRepository<Library, String> {

    boolean existsByPathIgnoreCase(String path);

    Optional<Library> findByPath(String path);
}

File: gameyfin/src/main/kotlin/de/grimsi/gameyfin/libraries/LibraryRepository.kt
package de.grimsi.gameyfin.libraries

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface LibraryRepository : JpaRepository<Library, Long> {
    @EntityGraph(attributePaths = ["games"])
    @Query("SELECT l FROM Library l ORDER BY function('RAND') LIMIT 1")
    fun findRandomLibrary(): Library?
}